homeassistant/components/aprilaire/climate.py

```diff
@@ -0,0 +1,360 @@
+
+        if self._coordinator.data.get(Attribute.DEHUMIDIFICATION_AVAILABLE) == 1:
+            features = features | ExtendedClimateEntityFeature.TARGET_DEHUMIDITY
+
+        if self._coordinator.data.get(Attribute.AIR_CLEANING_AVAILABLE) == 1:
+            features = features | ExtendedClimateEntityFeature.AIR_CLEANING
+
+        if self._coordinator.data.get(Attribute.VENTILATION_AVAILABLE) == 1:
+            features = features | ExtendedClimateEntityFeature.FRESH_AIR
+
+        features = features | ClimateEntityFeature.PRESET_MODE
+
+        features = features | ClimateEntityFeature.FAN_MODE
+
+        return features
+
+    @property
+    def current_humidity(self) -> int | None:
+        """Get current humidity."""
+        return self._coordinator.data.get(
+            Attribute.INDOOR_HUMIDITY_CONTROLLING_SENSOR_VALUE
+        )
+
+    @property
+    def target_humidity(self) -> int | None:
+        """Get current target humidity."""
+        return self._coordinator.data.get(Attribute.HUMIDIFICATION_SETPOINT)
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Get HVAC mode."""
+
+        if mode := self._coordinator.data.get(Attribute.MODE):
+            if hvac_mode := HVAC_MODE_MAP.get(mode):
+                return hvac_mode
+
+        return None
+
+    @property
+    def hvac_modes(self) -> list[HVACMode]:
+        """Get supported HVAC modes."""
+
+        if modes := self._coordinator.data.get(Attribute.THERMOSTAT_MODES):
+            if thermostat_modes := HVAC_MODES_MAP.get(modes):
+                return thermostat_modes
+
+        return []
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Get the current HVAC action."""
+
+        heating_equipment_status = self._coordinator.data.get(
+            Attribute.HEATING_EQUIPMENT_STATUS, 0
+        )
+
+        if heating_equipment_status > 0:
+            return HVACAction.HEATING
+
+        cooling_equipment_status = self._coordinator.data.get(
+            Attribute.COOLING_EQUIPMENT_STATUS, 0
+        )
+
+        if cooling_equipment_status > 0:
+            return HVACAction.COOLING
+
+        return HVACAction.IDLE
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Get current temperature."""
+        return self._coordinator.data.get(
+            Attribute.INDOOR_TEMPERATURE_CONTROLLING_SENSOR_VALUE
+        )
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Get the target temperature."""
+
+        hvac_mode = self.hvac_mode
+
+        if hvac_mode == HVACMode.COOL:
+            return self.target_temperature_high
+        if hvac_mode == HVACMode.HEAT:
+            return self.target_temperature_low
+
+        return None
+
+    @property
+    def target_temperature_step(self) -> float | None:
+        """Get the step for the target temperature based on the unit."""
+        return (
+            PRECISION_HALVES
```

Comment:
I think we only use this for precision, not for temperature steps

Reply:
@joostlek Fair point. There don't exist any constants for the steps as far as I can tell, so I replaced with `0.5` and `1` respectively in the interest of not misusing other constants.